real-main/app/handlers/appsync/handlers.py

```diff
@@ -401,6 +406,19 @@ def set_user_apns_token(caller_user, arguments, **kwargs):
     return caller_user.serialize(caller_user.id)
 
 
+@routes.register('Mutation.setUserDatingStatus')
+@validate_caller
+@update_last_client
+def set_user_dating_status(caller_user, arguments, **kwargs):
+    status = arguments.get('status')
+
+    user = user_manager.get_user(caller_user.id)
+    validate_dating_status_access_permission(user)
+
+    caller_user.set_dating_status(status)
```

Comment:
```suggestion
    try:
        caller_user.set_dating_status(status)
    except UserException as err:
        raise ClientException(str(err)) from err
```

`caller_user` has already been initalized from dynamo, no need to get it again.

And this validation logic, since it is dependent on the state of the user, should go within `set_dating_status()` (and raise a `UserException` if it fails).